### PR TITLE
support random seed for RA sampler

### PR DESCRIPTION
**Pull request by xiaohu2015**
This pr is about https://github.com/pytorch/vision/pull/5051
Two modification:
- support random seed for RA sampler
- fix import bug for RA sampler

cc @datumbox

### CODE REVIEW

File: references/classification/train.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import torchvision
 import transforms
 import utils
-from references.classification.sampler import RASampler
+from sampler import RASampler
```
Comment (datumbox) [inline references/classification/train.py:12]: Good catch.

File: references/classification/sampler.py [hunk 2]
```diff
@@ -15,7 +15,7 @@ class RASampler(torch.utils.data.Sampler):
     https://github.com/facebookresearch/deit/blob/main/samplers.py
     """
 
-    def __init__(self, dataset, num_replicas=None, rank=None, shuffle=True):
+    def __init__(self, dataset, num_replicas=None, rank=None, shuffle=True, seed=0):
```
Comment (datumbox) [inline references/classification/sampler.py:18]: I agree this might be useful but it's also a paradigm we don't do on TorchVision. Don't get me wrong, this approach is popular with other libraries it's just that setting seeds on constructors is something we don't typically do.

@fmassa thoughts?

Comment (xiaohu2015) [reply references/classification/sampler.py:18]: @datumbox The RA sampler is modified from torch.utils.data.DistributedSampler which also has a parameter `seed`. The default value of `seed` is 0, so it does not affect the behavior of RA samper if users don't explicitly set a seed.

File: references/classification/train.py (hunk 1 above)
Comment (datumbox) [reply references/classification/train.py:12]: I will need to patch this urgently as it is interfering with an investigation. See https://github.com/pytorch/vision/pull/4997#discussion_r765129140

Thanks for reporting @xiaohu2015

File: references/classification/sampler.py (hunk 2 above)
Comment (datumbox) [reply references/classification/sampler.py:18]: Fair enough. Since DistributedSampler supports it we can also have it.

**Review by datumbox — APPROVED** [review]
